core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -883,4 +929,24 @@ public static boolean canRewriteToMatchNone(SearchSourceBuilder source) {
         }
         return true;
     }
+
+    /*
+     * Rewrites the search request with a light weight rewrite context in order to fetch resources asynchronously
+     * The action listener is guaranteed to be executed on the search thread-pool
+     */
+    private void rewriteShardRequest(ShardSearchRequest request, ActionListener<ShardSearchRequest> listener) {
+        Rewriteable.rewriteAndFetch(request.getRewriteable(), indicesService.getRewriteContext(request::nowInMillis),
```

Comment:
would be nice to address the unchecked warning on this line. To be honest I tried that myself and I didn't succeed :)